cupy/core/core.pyx

```diff
@@ -1717,51 +1719,213 @@ cpdef ndarray array(obj, dtype=None, bint copy=True, order='K',
     elif hasattr(obj, '__cuda_array_interface__'):
         return array(_convert_object_with_cuda_array_interface(obj),
                      dtype, copy, order, subok, ndmin)
+    else:  # obj is sequence, numpy array, scalar or the other type of object
+        shape, elem_type, elem_dtype = _get_concat_shape(obj)
+        if shape is not None and shape[-1] != 0:
```

Comment:
Doesn't it cause index error if shape is `()`?

Reply:
`shape` is always non-empty tuple because a scalar value is *not* concatenatable and `_get_concat_shape()` always return `None` instead of `()`. I added `test_array_from_numpy_scalar` in fe85a30 to check this behavior:

```python
>>> a = np.array(0, dtype='b')
>>> a.ndim
0
>>> a.shape
()
>>> cupy.array(a)
array(0, dtype=int8)
>>> cupy.array([a])
array([0], dtype=int8)
```